### PR TITLE
soumyo/aws_ec2_vpn_connection_route

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by Nirbhay1997 — APPROVED** [review]
Please see if the singular resource is required or not

**Comment by IanMadd** [comment]
What's the difference between the two resources? It looks like they both test a single connection route.

**Comment by soumyo13** [comment]
> What's the difference between the two resources? It looks like they both test a single connection route.

In the singular resource, it gives the information of a specific route by filtering with vpn_connection_id.

In the plural resource, it fetch all the routes of the VPN connection.

File: docs/resources/aws_ec2_vpn_connection_routes.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+Use the `aws_ec2_vpn_connection_routes` InSpec audit resource to test properties of a plural AWS EC2 VPN Connection Route.
+
+The AWS::EC2::VPNConnectionRoute resource specifies a static route for a VPN connection between an existing virtual private gateway and a VPN customer gateway. The static route allows traffic to be routed from the virtual private gateway to the VPN customer gateway.
+
+## Syntax
+
+Ensure that VPN Connection ID exists.
+
+    describe aws_ec2_vpn_connection_routes do
```
Comment (IanMadd) [inline docs/resources/aws_ec2_vpn_connection_routes.md:16]: Does this require a VPN connection ID?

Comment (soumyo13) [reply docs/resources/aws_ec2_vpn_connection_routes.md:16]: No

Comment (IanMadd) [reply docs/resources/aws_ec2_vpn_connection_routes.md:16]: Is it an optional parameter?

Comment (soumyo13) [reply docs/resources/aws_ec2_vpn_connection_routes.md:16]: NO

Comment (IanMadd) [reply docs/resources/aws_ec2_vpn_connection_routes.md:16]: Then why is it on line 10 of libraries/aws_ec2_vpn_connection_routes.rb:
`describe aws_ec2_vpn_connection_routes(vpn_connection_id: 'VPNConnectionID') do`
Also lines 25 and 26?

Comment (soumyo13) [reply docs/resources/aws_ec2_vpn_connection_routes.md]: vpn_connection_id is required in the plural resource. I have added the same in the documentation.

**Review by IanMadd — APPROVED** [review]

**Pull request merged**